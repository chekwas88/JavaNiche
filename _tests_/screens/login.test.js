import React from 'react';
import { shallow } from 'enzyme';
import LoginScreen from '../../src/screens/Login';

describe('Test for Login component', () => {
  const wrapper = shallow(<LoginScreen />);
  test('should test for login ', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
describe('Text for login action', () => {
  it('should call onPress', () => {
    const wrapper = shallow(<LoginScreen />);
    const touchButton = wrapper
      .find('View')
      .find('View')
      .find('TouchableOpacity');
    const instance = wrapper.instance();
    instance.onPressHandler();
    const mockOnPress = (instance.onPressHandler = jest.fn());
    touchButton.simulate('onPress', mockOnPress());
    expect(mockOnPress).toHaveBeenCalled();
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
