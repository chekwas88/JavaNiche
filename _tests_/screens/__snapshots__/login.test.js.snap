// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Login component should test for login  1`] = `ShallowWrapper {}`;
